### ISSUE TITLE
[rllib] DDPG opensim env issue

### ISSUE THREAD

**Issue by whikwon** [ray-project/ray#2648]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: ubuntu 16.04
- **Ray installed from (source or binary)**: source
- **Ray version**: 0.5.0
- **Python version**: 3.6.3
- **Exact command to reproduce**:

 

### Describe the problem

Executing DDPG in  `Pendulum-v0` environment has no problem. 

However, when i execute DDPG in  `Prosthetics` environment, error occurs during gradient calculation. 
Do I have to add some additional code in custom environment to solve this problem? 

![image](https://user-images.githubusercontent.com/30768596/44073432-d325c11a-9fce-11e8-839b-66ade9fb9d02.png)

### Source code / logs

```python
import ray 
import ray.tune.registry import register_env
from ray.rllib.agents import ddpg
from osim.env import ProstheticsEnv

def env_creator(env_config):
    return ProstheticsEnv(False)

register_env("prosthetics", env_creator)
ray.init("<head-ip:port>")

agent = ddpg.DDPGAgent(env="prosthetics", config={"num_workers": 140, "gpu": True})

for i in range(1000):
    result = agent.train()
    print(result)
    if i % 10 == 0:
        agent.save()

```

**Comment by whikwon** [comment]
@ericl I tried `print([type(s) for s in samples])` and got all the `ray.rllib.evaluation.sample_batch.SampleBatch` stuff.

**Comment by ericl** [comment]
That's weird, and it definitely crashes on the next line with the type of s being MultiAgentBatch?

It would be great to get a smaller reproduction case that doesn't require osim, I tried running the above and could not reproduce the crash on python 3.6 (though I didn't fully install osim, just used a similar env).

**Comment by whikwon** [comment]
Yeah, I tried `Pendulum-v0` and it worked. I think it doesn't work well only for the osim environment.
